app/views/govuk_publishing_components/components/_intervention.html.erb

```diff
@@ -0,0 +1,24 @@
+<%
+  local_assigns[:margin_bottom] ||= 8
+  local_assigns[:margin_bottom] = 8 if local_assigns[:margin_bottom] > 9
+
+  shared_helper = GovukPublishingComponents::Presenters::SharedHelper.new(local_assigns)
+
+  css_classes = %w[gem-c-intervention]
+  css_classes << (shared_helper.get_margin_bottom)
+
+  aria_attributes = {label: 'Intervention'}
+
+  variant = request.headers["HTTP_GOVUK_ABTEST_STARTABUSINESSSEGMENT"]
+  is_sab_page = request.headers["HTTP_GOVUK_ABTEST_ISSTARTABUSINESSPAGE"] == "true"
+  should_show_intervention = variant == "B" && is_sab_page
```

Comment:
Having the A/B test logic inside the component feels a bit icky - could this logic be moved into the application with the component itself completely placed inside the `if` statement inside the application's view?

For example, in the application the logic could be:
```erb
  <%= render("govuk_publishing_components/components/intervention") if should_show_intervention %>
```

Reply:
We're running the A/B test across three different apps (and probably in multiple views of those different apps) so it seemed simpler to have the logic in one place, even if it is a bit unconventional - saves having three implementations and three sets of tests and the associated increase in maintenance/potential for problems

Reply:
I'm going to disagree here. It's simpler to have the business logic in the controller of the application - all logic is in that one place in the application and not spread out over views, controllers, gems, and so on. Trying to trace exactly where a decision to show or hide a component becomes a lot harder if the logic sits outside of the application, and especially if it sits outside of Rails expected conventions.

I agree it's easier in the short term to drop this into the components gem, but I think it'll create more problems that it'll solve. The components gem is supposed to be for components - if you really need the logic in one place, then it shouldn't be in the components gem.

Reply:
Thanks, fair enough. We've now removed application logic from the component.